File: backend/src/modloader/neoforge.ts
import fse from "fs-extra"
import { Forge } from "./forge.js";

export class NeoForge extends Forge{
constructor(minecraft:string,loaderVersion:string,path:string){
    super(minecraft,loaderVersion,path); //子承父业
}

async setup(){
    await this.installer();
    await this.library();
    await this.install();
}

async installer(){
    const res = (await this.got.get(`neoforge/version/${this.loaderVersion}/download/installer.jar`)).rawBody;
    await fse.outputFile(`${this.path}/forge-${this.minecraft}-${this.loaderVersion}-installer.jar`,res);
}
}

File: backend/modloader/neoforge.go
package modloader

import (
	"dex/backend/utils"
	"path/filepath"
)

func NewNeoForge(mcv, mlv, path string) *NeoForge {
	return &NeoForge{
		Forge: NewForge(mcv, mlv, path),
	}
}

type NeoForge struct {
	*Forge
}

func (super *NeoForge) Installer() error {
	var url string
	var expectedHash string

	if utils.GlobalConfig.GetConfigValue("mirror.bmclapi") == true {
		url = "https://bmclapi2.bangbang93.com" + "/neoforge/version/" + super.loaderVersion + "/download/installer.jar"
	} else {
		url = "https://maven.neoforged.net/releases/net/neoforged/neoforge/" + super.loaderVersion + "/neoforge-" + super.loaderVersion + "-installer.jar"
	}
	
	filePath := filepath.Join(super.path, "forge-"+super.minecraft+"-"+super.loaderVersion+"-installer.jar")

	return nil
}
